src/connection/class-term-objects.php

```diff
@@ -38,10 +37,10 @@ public static function register_connections() {
 					]
 				),
 				'resolve'        => function ( $source, $args, $context, $info ) {
-					$taxonomies = isset( $args['where']['taxonomies'] ) && is_array( $args['where']['taxonomies'] ) ? $args['where']['taxonomies'] : \WPGraphQL::get_allowed_taxonomies();
-					$resolver   = new TermObjectConnectionResolver( $source, $args, $context, $info, array_values( $taxonomies ) );
-					$connection = $resolver->get_connection();
-					return $connection;
+					$taxonomies = isset( $args['where']['taxonomies'] ) && is_array( $args['where']['taxonomies'] )
+						? $args['where']['taxonomies']
+						: \WPGraphQL::get_allowed_taxonomies();
+					return DataSource::resolve_term_objects_connection( $source, $args, $context, $info, array_values( $taxonomies ) );
```

Comment:
I'm hesitant to make these changes. 

The new ConnectionResolver classes are flexible and intended to allow extension authors the ability to use them to resolve custom connections more easily. 

For example, let's say a custom connection were created to display a connection from Posts of one Post Type to Posts of another Post Type, and the IDs of the connected posts were stored in meta. 

A custom connection could be registered like so: 

```
register_graphql_connection([
  'fromType' => 'Post',
  'toType' => 'Post',
  'fromFieldName' => 'relatedPosts',
  'connectionTypeName' => 'RelatedPosts', 
  'resolve' => function( $source, $args, $context, $info ) {
      $related_ids = get_post_meta( $source->ID, 'related_posts', true );
      if ( ! empty( $related_ids ) ) {
          $resolver = new Post_Object_Connection_Resolver( $source, $args, $context, $info );
          $resolver->setQueryArg( 'post__in', $related_posts );
          return $resolver->get_connection();
      }
      return null;
  }
]);
```

I'm actually inclined to make _more_ resolvers throughout the graph look like _this_ and cleanup some deep internals within connection resolvers. 

For example: https://github.com/wp-graphql/wp-graphql/blob/develop/src/Data/Connection/PostObjectConnectionResolver.php#L244-L266

Instead of this switch statement living within the ConnectionResolver, it could be more explicitly defined for the resolvers that need this context. 

Like here, for example: https://github.com/wp-graphql/wp-graphql/blob/develop/src/Data/Connection/PostObjectConnectionResolver.php#L246

This line sets the `post_parent` argument if the source of the connection is a Post. 

This screws up connections like the custom one above, where it would be a Post->Post connection, but not a parent/child connection. If this argument `post_parent=>0` were defined only in the connection resolvers that _need_ it, it would allow custom connections to be registered and resolved with less friction. 

Resolving with `DataSource::resolve_**_connection()` calls doesn't provide this flexibility, nor does it document by example for extension authors how they can utilize the connection resolver classes to "easily" resolve their custom connections.

Reply:
Then why not make the `DataSource::resolve_**_connection()` return an instance of the resolver class instead? The naming convention can even change to `DataSource::**_connections()`. This way you can do something like this
```
register_graphql_connection([
  'fromType' => 'Post',
  'toType' => 'Post',
  'fromFieldName' => 'relatedPosts',
  'connectionTypeName' => 'RelatedPosts', 
  'resolve' => function( $source, $args, $context, $info ) {
      $related_ids = get_post_meta( $source->ID, 'related_posts', true );
      if ( ! empty( $related_ids ) ) {
          $resolver = DataSource::post_object_connections( $source, $args, $context, $info );
          $resolver->setQueryArg( 'post__in', $related_posts );
          return $resolver->get_connection();
      }
      return null;
  }
]);
```